src/tests/controllers/UserController.test.ts

```
@@ -42,22 +42,30 @@ describe("UserController", () => {
+    };
+
+    userService.createUser.mockResolvedValue(createdUser as unknown as User);
     req.body = userData;
 
     await userController.createUser(req as Request, res as Response);
 
-    const mutableUser: Partial<User> = { ...createdUser };
-    delete mutableUser.password;
+    const mutableUser = { ...createdUser };
+    // Use type assertion to delete password property
+    delete (mutableUser as Partial<typeof mutableUser> & { password?: string }).password;
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)